web/src/product/ProductForm.js

```diff
@@ -21,50 +21,223 @@ async function addProduct(token, url, data) {
+        });
+        return response.json();
+    } catch (e) {
+        console.log(e);
+    }
+}
+
+async function getAllProduct() {
+    const url = "http://localhost:4000/api/v1/products/";
+    try {
+        const response = await fetch(url, {
+            method: 'GET', // *GET, POST, PUT, DELETE, etc.
+            mode: 'cors', // no-cors, *cors, same-origin
+            cache: 'no-cache', // *default, no-cache, reload, force-cache, only-if-cached
+            credentials: 'same-origin', // include, *same-origin, omit
+            headers: {
+                'Authorization': 'Bearer ' + localStorage.token,
+                'Content-Type': 'application/json'
+            },
+            redirect: 'follow', // manual, *follow, error
+            referrerPolicy: 'no-referrer', // no-referrer, *no-referrer-when-downgrade, origin, origin-when-cross-origin, same-origin, strict-origin, strict-origin-when-cross-origin, unsafe-url
+        });
+        return response.json();
+    } catch (e) {
+        console.log(e);
+    }
+}
+
+async function updateProduct(url, data) {
+    try {
+        const response = await fetch(url, {
+            method: 'PATCH', // *GET, POST, PUT, DELETE, etc.
+            mode: 'cors', // no-cors, *cors, same-origin
+            cache: 'no-cache', // *default, no-cache, reload, force-cache, only-if-cached
+            credentials: 'same-origin', // include, *same-origin, omit
+            headers: {
+                'Authorization': 'Bearer ' + localStorage.token,
+                'Content-Type': 'application/json'
+            },
+            redirect: 'follow', // manual, *follow, error
+            referrerPolicy: 'no-referrer', // no-referrer, *no-referrer-when-downgrade, origin, origin-when-cross-origin, same-origin, strict-origin, strict-origin-when-cross-origin, unsafe-url
+            body: JSON.stringify(data) // body data type must match "Content-Type" header
+        });
+        return response.json();
+    } catch (e) {
+        console.log(e);
+    }
+}
+
+
 const ProductForm = (props) => {
-    const formik = useFormik({
+    const formikAdd = useFormik({
         initialValues: {
             name: '',
             tag: ''
         },
         onSubmit: async (values) => {
-            const res = await addProduct(props.token, "http://localhost:4000/api/v1/products/add", values);
+            const res = await addProduct("http://localhost:4000/api/v1/products/add", values);
             if (res.status === "fail") {
-                formik.errors.submit = res.message;
+                formikAdd.errors.submit = res.message;
             } else {
                 console.log("Is ok");
+                alert("The product: " + JSON.stringify(res.data.name) + " has correctly been added");
             }
         }
     });
 
+    const formikGet = useFormik({
```

Comment:
Do we really expect user to know the id of a product ? Not sure this is useful.
Better to have a list that display all products

Reply:
true